File: src/main/java/scrumweb/dto/projectfield/ProjectFieldsCollector.java
package scrumweb.dto.projectfield;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.stereotype.Component;
import scrumweb.common.FieldsExtractor;

import java.util.HashSet;
import java.util.Set;

@NoArgsConstructor
@AllArgsConstructor
@Component
@Getter @Setter
public class ProjectFieldsCollector implements FieldsExtractor<ProjectFieldDto> {

    private Set<CheckBoxContainerDto> checkBoxContainerDtos;
    private Set<InputFieldDto> inputFieldDtos;
    private Set<ListElementsContainerDto> listElementsContainerDtos;
    private Set<RadioButtonContainerDto> radioButtonContainerDtos;
    private Set<TextAreaDto> textAreaDtos;

    @Override
    public Set<ProjectFieldDto> extractFields() {
        Set<ProjectFieldDto> projectFieldDtos = new HashSet<>();
        projectFieldDtos.addAll(checkBoxContainerDtos);
        projectFieldDtos.addAll(inputFieldDtos);
        projectFieldDtos.addAll(listElementsContainerDtos);
        projectFieldDtos.addAll(radioButtonContainerDtos);
        projectFieldDtos.addAll(textAreaDtos);
        return projectFieldDtos;
    }
}

File: src/frontend/src/app/model/project-fields/ProjectFieldsCollector.ts
import { CheckBoxContainerDto } from './CheckBoxContainerDto';
import { InputFieldDto } from './InputFieldDto';
import { ListElementsContainerDto } from './ListElementsContainerDto';
import { RadioButtonContainerDto } from './RadioButtonContainerDto';
import { TextAreaDto } from './TextAreaDto';

export class ProjectFieldsCollector {
    public checkBoxContainerDtos: Array<CheckBoxContainerDto> = [];
    public inputFieldDtos: Array<InputFieldDto> = [];
    public listElementsContainerDtos: Array<ListElementsContainerDto> = [];
    public radioButtonContainerDtos: Array<RadioButtonContainerDto> = [];
    public textAreaDtos: Array<TextAreaDto> = [];
}
